src/System.Private.CoreLib/shared/System/Text/UnicodeEncoding.cs

```diff
@@ -695,14 +699,14 @@ internal override unsafe int GetByteCount(char* chars, int count, EncoderNLS enc
                 if (ch == 0)
                 {
                     // No fallback, maybe we can do it fast
-#if !NO_FAST_UNICODE_LOOP
+#if FASTLOOP
                     // If endianess is backwards then each pair of bytes would be backwards.
                     if ( (bigEndian ^ BitConverter.IsLittleEndian) && 
-#if BIT64           // 64 bit CPU needs to be long aligned for this to work, 32 bit CPU needs to be 32 bit aligned
-                        (unchecked((long)chars) & 7) == 0 && (unchecked((long)bytes) & 7) == 0 &&
```

Comment:
>(unchecked((long)bytes) & 7) == 0  [](start = 61, length = 34)

we don't care about the bytes alignment here?

Reply:
The case of mis-aligned `bytes` is taken care of by using `Unsafe.WriteUnaligned` for writing to them.